homeassistant/components/climate/__init__.py

```diff
@@ -449,8 +480,12 @@ class ClimateDevice(Entity):
     @property
     def state(self):
         """Return the current state."""
+        if self.is_on is False:
+            return STATE_OFF
         if self.current_operation:
             return self.current_operation
+        if self.is_on:
+            return STATE_ON
         return STATE_UNKNOWN
```

Comment:
This should preferably return `None` as we want the `Entity` base class to handle that case, ie set `STATE_UNKNOWN`.